src/client.rs

```diff
@@ -325,6 +328,105 @@ impl Client {
         )
         .await
     }
+
+    /// Wait until Meilisearch processes a [Task], and get its status.
+    ///
+    /// `interval` = The frequency at which the server should be polled. Default = 50ms
+    /// `timeout` = The maximum time to wait for processing to complete. Default = 5000ms
+    ///
+    /// If the waited time exceeds `timeout` then an [Error::Timeout] will be returned.
+    ///
+    /// See also [Task::wait_for_completion].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use meilisearch_sdk::{client::*, document, indexes::*, tasks::Task};
+    /// # use serde::{Serialize, Deserialize};
+    /// #
+    /// # #[derive(Debug, Serialize, Deserialize, PartialEq)]
+    /// # struct Document {
+    /// #    id: usize,
+    /// #    value: String,
+    /// #    kind: String,
+    /// # }
+    /// #
+    /// # impl document::Document for Document {
+    /// #    type UIDType = usize;
+    /// #
+    /// #    fn get_uid(&self) -> &Self::UIDType {
+    /// #        &self.id
+    /// #    }
+    /// # }
+    /// #
+    /// # futures::executor::block_on(async move {
+    /// let client = Client::new("http://localhost:7700", "masterKey");
+    /// let movies = client.index("movies_wait_for_task");
+    ///
+    /// let task = movies.add_documents(&[
+    ///     Document { id: 0, kind: "title".into(), value: "The Social Network".to_string() },
+    ///     Document { id: 1, kind: "title".into(), value: "Harry Potter and the Sorcerer's Stone".to_string() },
+    /// ], None).await.unwrap();
+    ///
+    /// let status = client.wait_for_task(task, None, None).await.unwrap();
+    ///
+    /// assert!(matches!(status, Task::Succeeded { .. }));
+    /// # movies.delete().await.unwrap().wait_for_completion(&client, None, None).await.unwrap();
+    /// # });
+    /// ```
+    pub async fn wait_for_task(
+        &self,
+        task_id: impl AsRef<u64>,
+        interval: Option<Duration>,
+        timeout: Option<Duration>,
+    ) -> Result<Task, Error> {
+        let interval = interval.unwrap_or_else(|| Duration::from_millis(50));
+        let timeout = timeout.unwrap_or_else(|| Duration::from_millis(5000));
+
+        let mut elapsed_time = Duration::new(0, 0);
+        let mut task_result: Result<Task, Error>;
+
+        while timeout > elapsed_time {
+            task_result = self.get_task(&task_id).await;
+
+            match task_result {
+                Ok(status) => match status {
+                    Task::Failed { .. } | Task::Succeeded { .. } => {
+                        return self.get_task(task_id).await;
+                    }
+                    Task::Enqueued { .. } | Task::Processing { .. } => {
+                        elapsed_time += interval;
+                        async_sleep(interval).await;
+                    }
+                },
+                Err(error) => return Err(error),
+            };
+        }
+
+        Err(Error::Timeout)
+    }
+
+    /// # Example
```

Comment:
```suggestion
    /// Get a task from its `uid`.
    ///
    /// # Example
```

Reply:
Kero already suggested the same change 😁